development-docs/TESTING.md

```diff
@@ -0,0 +1,186 @@
+- [Package Structure](#package-structure)
+- [Approaches](#approaches)
+- [Test Phases](#test-phases)
+- [Available Test Groups](#available-test-groups)
+- [Environment Variables](#environment-variables)
+- [Running single test class](#running-single-test-class)
+
+<!-- /TOC -->
+
+## Pre-requisites
+
+To run any tests you need build whole project. You can achieve that with following command  `mvn clean install -DskipTests` 
+in root directory of the project.
+
+The next requirement is to have [docker](https://docs.docker.com/get-docker/) installed, because we use test-containers. 
+In case of interest you can check our implementation of Strimzi `Kafka` container [here](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test-container).
+
+## Package Structure
+
+You can find tests inside `src.test.java` package. Moreover we have the auxiliary classes and the most notable one are:
+ 
+- `src.test.java.io.strimzi.kafka.bridge.utils` = static methods
+- `src.test.java.io.strimzi.kafka.bridge.clients` = clients for testing overall communication of the `Kafka Bridge`
+- `src.test.java.io.strimzi.kafka.bridge.facades` = encapsulation of the standalone `Kafka` and `AdminClient` instance
+
+Furthermore we divide our test suite on two separate parts:
+
+1. `src.test.java.io.strimzi.kafka.bridge.http` = package, where you can find all test cases related to the `Http Kafka Bridge`
+2. `src.test.java.io.strimzi.kafka.bridge.amqp` = package, where you can find all tests cases related to the `Amqp Kafka Bridge`
+
+## Approaches
+
+There are four approaches how you can verify that `Kafka Bridge` is without any bug. Before I list all approaches 
+the following  terminology is needed.
+
+#### In-memory 
+`Kafka` = 
+- for `Http Kafka Bridge` we use [Strimzi Kafka container]([here](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test-container).)
+- for `Amqp Kafka Bridge` we use [KafkaFacade](https://github.com/strimzi/strimzi-kafka-bridge/blob/master/src/test/java/io/strimzi/kafka/bridge/facades/KafkaFacade.java) 
+
+`Kafka Bridge` =
+ -  we use for deployment `Http Kafka Bridge` and `Amqp Kafka Bridge` the [Vert.x](https://vertx.io/) 
+
+#### Standalone 
+`Kafka` 
+ - we use latest version of the `Kafka` and you can start the `Kafka` instance by the `zookeeper-server-start.sh` and
+`kafka-server-start.sh`, which can be for instance download from [this](https://kafka.apache.org/downloads) site
+
+`Kafka Bridge` 
+ - start `Kafka Bridge` instance using bash script located `target/kafka-bridge-{version}/kafka-bridge-{version}/bin/kafka_bridge_run.sh` 
+with following command `kafka_bridge_run.sh --config-file ../config/application.properties`
+ 
+----  
+The approaches are as follows:
+
+1. Using in-memory `Kafka` and in-memory `Kafka Bridge`
+2. Using standalone `Kafka` and standalone `Kafka Bridge`
+3. Using in-memory `Kafka` and standalone `Kafka Bridge`
+4. Using standalone `Kafka` and in-memory `Kafka Bridge`
```

Comment:
is it feasible to have a table 2 x 2 checking what is supported and what is not?